File: Swift-C++-Testsuite/test/arithmetic-operators/Multiplication/Swift/OverflowMul.swift
@main
struct MyMain {

    static func main() {
        // This call would crash with a normal add.
        // returns -9223372036854775808
        let a = mulWithOverflow(x: Int.max)
    }

    // The & sign turns an mulition to an mulition without overflow
    static func mulWithOverflow(x: Int) -> Int {
        var tmp = x &* 2
        return tmp
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/OverflowMul.cpp

int MyMain() 
{
    // This call would crash with a normal mul.
    // returns -9223372036854775808
    int a = mulWithOverflow(INT_MAX);

    // The & sign turns an mulition to an mulition without overflow
    int mulWithOverflow(int x) 
    {
        int tmp = x &* 2;
        return tmp;
    }
    
    return 0;
}